src/server/clientiface.cpp

```diff
@@ -339,6 +339,15 @@ void RemoteClient::GetNextBlocks (
 				if (d >= d_opt && block->isAir())
 						continue;
 			}
+
+			const bool want_emerge = !block || !block->isGenerated();
+
+			// if the block is already in the emerge queue we don't have to check again
+			if (want_emerge && emerge->isBlockInQueue(p)) {
+				nearest_emerged_d = d;
```

Comment:
there's a bug here actually, the `if (nearest_emerged_d == -1)` check is missing
so the variable will contain the *furthest* position in the end